### PR TITLE
#1135 New SupplierRequisitionsPage - #1

### LINKED ISSUE #1135: Refactor SupplierRequisitionsPage to use new DataTable
## Is your feature request related to a problem? Please describe.
Need to use the new `DataTable` in `SupplierRequisitionsPage`.

## Describe the solution you'd like

- A more or less complete refactor, using the first refactored page `CustomerInvoicePage` as a guide.
- This will require using hooks and the new DataTable `component`.

## Describe alternatives you've considered

N/A

## Additional context

This is a placeholder issue. Please investigate and compare the page with `CustomerInvoicePage`. Each page being refactored may have their own quirks. The tissue is encouraged to describe any additional problems and differences with the particular page here, to find a solution which encourages further extensibility and code-reuse with other pages, if possible.

### PR DESCRIPTION
**Pull request by joshxg**
#### EPIC: #1043 

Fixes #1135 

## Change summary

- Refactors `SupplierRequisitionPage` to use the new `DataTable`

## Testing

SEE: #1043 

### Related areas to think about

- Currently will not refresh when navigating back to the page.

### CODE REVIEW

**Comment by joshxg** [comment]
Issue #1195 is made for fixes needing to be done related to this PR. Have found the entered by column key in `sortDataBy` wasn't merged and crashes, so far

File: src/pages/SupplierRequisitionsPage.js [hunk 1]
```diff
@@ -1,271 +1,198 @@
...
+    newPageTopLeftSectionContainer,
+    newPageTopRightSectionContainer,
+    searchBar,
+  } = newPageStyles;
+  return (
+    <DataTablePageView>
+      <View style={newPageTopSectionContainer}>
+        <View style={newPageTopLeftSectionContainer}>
+          <SearchBar
+            onChange={value => debouncedDispatch(filterData(value))}
```
Comment (mark-prins) [inline src/pages/SupplierRequisitionsPage.js]:
```suggestion
            onChange={handleSearch}
```

with

```
const handleSearch value => debouncedDispatch(filterData(value));
```

declared somewhere outside of the `render()` method.
Would prevent creation of a method within render.

File: src/pages/SupplierRequisitionsPage.js [hunk 2]
```diff
@@ -1,271 +1,198 @@
...
+        extraData={dataState}
+        renderRow={renderRow}
+        renderHeader={renderHeader}
+        keyExtractor={recordKeyExtractor}
+        getItemLayout={getItemLayout}
+      />
+      <BottomConfirmModal
+        isOpen={hasSelection}
+        questionText={modalStrings.remove_these_items}
+        onCancel={() => dispatch(deselectAll())}
```
Comment (mark-prins) [inline src/pages/SupplierRequisitionsPage.js]: also probably not required, but could declare these (3x anonymous methods) outside of `render`

File: src/pages/SupplierRequisitionsPage.js [hunk 3]
```diff
@@ -1,271 +1,198 @@
...
+  );
+
+  const renderButtons = () => {
+    const { verticalContainer, topButton } = globalStyles;
+    return (
+      <View style={verticalContainer}>
+        <PageButton
+          style={topButton}
+          text={buttonStrings.new_requisition}
+          onPress={() => dispatch(openBasicModal(NEW_REQUISITON))}
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/pages/SupplierRequisitionsPage.js]: Didn't know if this was over kill, or not, heh.  Will do this for all pages

File: src/pages/SupplierRequisitionsPage.js [hunk 4]
```diff
@@ -1,271 +1,198 @@
...
+    sortBy: 'serialNumber',
+    isAscending: true,
+    modalKey: '',
+    hasSelection: false,
+    currentUser,
+  });
+
+  const { data, dataState, sortBy, isAscending, columns, modalKey, hasSelection } = state;
+
+  const usingPrograms = useCallback(getAllPrograms(Settings, UIDatabase).length > 0, []);
```
[the remark this answers is not in the document]
Comment (joshxg) [reply src/pages/SupplierRequisitionsPage.js]: you're right - it does function as expected. But taking another look, it .. shouldn't. I would think it should be `() => ...`, and `useMemo` would need to be `() => () => ...` - but adding `() => ..` stops it working. I'm confused now also - I think swapping this to `useMemo` is more explicit about the intended use of just memoizing the bool anyway!

**Pull request merged**